### ISSUE TITLE
SSL support in cassandra thrift client

### ISSUE THREAD

**Issue by grozavpava** [thinkaurelius/titan#629]
This is actually a feature request.

Enabling client-to-node SSL connections from Titan to Cassandra should be possible.
Cassandra already supports this, and Apache Thrift has this functionality since 0.6.x (0.4.2 is using 0.7.0)

I understand that for future releases CQL/java driver is the preferred way to connect to Cassandra, but i think the thrift client can be enhanced also.

I hope to get some time to work and this and create a pull request.

**Comment by frankjl** [comment]
I came across this recently too, and the lack of client-to-node SSL was a deal breaker for me.  I ended up patching com.thinkaurelius.titan.diskstorage.cassandra.astyanax.AstyanaxStoreManager with the following snippet @ line 480:

``` java
if (config.getBoolean("ssl.enabled", false)) {
        String sslTruststore = config.getString("ssl.truststore.location");
        String sslTruststorePassword = config.getString("ssl.truststore.password");
        cpool.setSSLConnectionContext(new SSLConnectionContext(sslTruststore, sslTruststorePassword));
}
```

Doing so let me define 3 more properties in my Titan configuration:

```
storage.ssl.enabled=true
storage.ssl.truststore.location=<<path to truststore>>
storage.ssl.truststore.password=<<truststore password
```

After this I was able to connect to my Cassandra cluster through Titan.  I'd be happy to fork & apply these changes if someone finds them helpful.

**Comment by dalaro** [comment]
@frankjl please do fork and open a pull request against master. Thanks!

**Comment by xedin** [comment]
@dalaro @mbroecheler Pull is at https://github.com/thinkaurelius/titan/pull/655 , supports both Astyanax and Thrift.

**Comment by ppeddi** [comment]
This is a deal breakers for us. If one of the committer could merge into mainline and release it in the next version, that would be very helpful for us.

**Comment by dalaro** [comment]
This has been already merged into the titan05 branch via #655.  It's in the 0.5.0-M1 preview release and will be present in 0.5.0 final and beyond.

We haven't backported #655 to master/0.4.x, but there's no fundamental reason we couldn't do that (PRs welcome).
